plugin/src/main/java/org/opensearch/ml/sdkclient/DDBOpenSearchClient.java

```diff
@@ -155,16 +200,140 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(DeleteDat
         }), executor);
     }
 
+    /**
+     * DDB data needs to be synced with opensearch cluster. {@link RemoteClusterIndicesClient} will then be used to
+     * search data in opensearch cluster.
+     *
+     * @param request
+     * @param executor
+     * @return Search data object response
+     */
     @Override
     public CompletionStage<SearchDataObjectResponse> searchDataObjectAsync(SearchDataObjectRequest request, Executor executor) {
-        // TODO will implement this later.
-
-        return null;
+        return this.remoteClusterIndicesClient.searchDataObjectAsync(request, executor);
     }
 
     private String getTableName(String index) {
         // Table name will be same as index name. As DDB table name does not support dot(.)
         // it will be removed from name.
         return index.replaceAll("\\.", "");
     }
+
```

Comment:
All these static methods should probably be in a util class, in which case they might be public anyway.

Reply:
Good suggestion. Will update.